assets/js/base/components/cart-checkout/totals/coupon/index.tsx

```diff
@@ -63,81 +58,89 @@ export const TotalsCoupon = ( {
 		if ( currentIsLoading.current !== isLoading ) {
 			if ( ! isLoading && couponValue && ! validationError ) {
 				setCouponValue( '' );
+				setdisplayCouponForm( false );
+				setdisplayAddCouponLink( true );
 			}
 			currentIsLoading.current = isLoading;
 		}
 	}, [ isLoading, couponValue, validationError ] );
 
 	return (
-		<Panel
-			className="wc-block-components-totals-coupon"
-			hasBorder={ false }
-			initialOpen={ initialOpen }
-			title={
-				<Label
-					label={ __(
-						'Coupon code',
-						'woo-gutenberg-products-block'
-					) }
+		<>
+			{ displayAddCouponLink && (
+				// eslint-disable-next-line jsx-a11y/anchor-is-valid
+				<a
+					role="button"
+					href="#"
```

Comment:
Nice work adapting this functionality. I noticed that the initial implementation was not accessible so I'm proposing to improve this in your implementation. If you would try to test this branch with the screen reader you will see that it is not a good experience.

We'll need to tweak this implementation to work well for screen readers as well.
Looking at the current implementation there are two issues:
- the anchor is not pointing to the hidden element; `href="#"` should be avoided, as it's not accessible. 
- the hidden element is not included in the page, making the whole area not screen reader friendly

Actually, the href of this link is the form (or the div container of the form, depending on what is hidden), so ideally the href should be "#id-of-the-from". Where <div id="id-of-the-from" ../>
The hidden content should be rendered, but not present on the page. I think there were some CSS classes in the project to hide content outside of the screen readers. When the user clicks the link the CSS class should be removed and the content should appear. 

We'd need to also make sure there is no issue with validation in this case, as in the form is visually hidden but still we run validation on it.

Reply:
That's actually a better approach. Thank you, I have made the changes! 👍🏻